include/zephyr/bluetooth/mesh/cdb.h

```diff
@@ -123,14 +124,19 @@ void bt_mesh_cdb_iv_update(uint32_t iv_index, bool iv_update);
  *  @param num_elem Number of elements that the node has.
  *  @param net_idx NetIdx that the node was provisioned to.
  *
- *  @return The new node or NULL if it cannot be allocated.
+ *  @return The new node or NULL if CDB has already allocated @ref CONFIG_BT_MESH_CDB_NODE_COUNT
+ *  nodes, or reached the end of the unicast address range, or if @c addr is non-zero and less than
+ *  the lowest available address.
```

Comment:
Since you expanded on this: also returns `NULL` for a colliding address even if that address is > lowest available address.